src/value/jink_constant.rs

```diff
@@ -0,0 +1,335 @@
+            ))
+        }
+    }
+}
+
+impl FromInstance for JinkChar {
+    fn from_instance(i: &Instance) -> Self {
+        use std::mem::{size_of, transmute};
+
+        unsafe {
+            JinkChar::from(transmute::<[u8; size_of::<char>()], char>(
+                TryFrom::try_from(i.data()).unwrap(),
+            ))
+        }
+    }
+}
+
+impl FromInstance for JinkString {
+    fn from_instance(i: &Instance) -> Self {
+        // unchecked is safe because this instance came from a utf8 string in ToInstance
+        unsafe { JinkString::from(String::from_utf8_unchecked(i.data().to_vec())) }
+    }
+}
+
+// impl FromInstance for JinkFloat {
+//     fn from_instance(i: &Instance) -> Self {
+//         use std::mem::{transmute, size_of};
+//
+//         JinkFloat::from(transmute::<[u8; size_of::<f64>()], f64>(i.data()))
+//     }
+// }
+//
+// impl FromInstance for JinkBool {
+//     fn from_instance(i: &Instance) -> Self {
+//         use std::mem::{transmute, size_of};
+//
+//         JinkBool::from(transmute::<[u8; size_of::<bool>()], bool>(i.data()))
+//     }
+// }
+//
+// impl FromInstance for JinkChar {
+//     fn from_instance(i: &Instance) -> Self {
+//         use std::mem::{transmute, size_of};
+//
+//         JinkChar::from(transmute::<[u8; size_of::<char>()], char>(i.data()))
+//     }
+// }
+//
+// impl FromInstance for JinkString {
+//     fn from_instance(i: &Instance) -> Self {
+//         use std::mem::{transmute, size_of};
+//
+//         JinkString::from(transmute::<[u8; size_of::<String>()], String>(i.data()))
+//     }
+// }
+
+impl Value for JinkConstant<i64> {
+    fn do_op(&self, other: &Self, op: Operator) -> Result<Instance, JinkoError> {
+        match op {
+            Operator::Add => Ok(JinkConstant::from(self.0 + other.0).to_instance()),
+            Operator::Sub => Ok(JinkConstant::from(self.0 - other.0).to_instance()),
+            Operator::Mul => Ok(JinkConstant::from(self.0 * other.0).to_instance()),
+            Operator::Div => Ok(JinkConstant::from(self.0 / other.0).to_instance()),
+            _ => self.no_op(other, op),
+        }
+    }
+}
+
+// FIXME: Avoid this copy paste, find a better/cleaner way to do it
+impl Value for JinkConstant<f64> {
+    fn do_op(&self, other: &Self, op: Operator) -> Result<Instance, JinkoError> {
+        match op {
+            Operator::Add => Ok(JinkConstant::from(self.0 + other.0).to_instance()),
+            Operator::Sub => Ok(JinkConstant::from(self.0 - other.0).to_instance()),
+            Operator::Mul => Ok(JinkConstant::from(self.0 * other.0).to_instance()),
+            Operator::Div => Ok(JinkConstant::from(self.0 / other.0).to_instance()),
+            _ => self.no_op(other, op),
+        }
+    }
+}
+
+impl<T> From<T> for JinkConstant<T> {
+    fn from(rust_value: T) -> Self {
+        JinkConstant(rust_value)
+    }
+}
+
+impl From<&str> for JinkConstant<String> {
+    fn from(s: &str) -> Self {
+        JinkConstant(s.to_string())
+    }
+}
```

Comment:
I think this file was added by mistake :D

Reply:
Could you elaborate ? This is due to the merge of the cleanup ?

Reply:
Yes, this is the old version of the file. You can see that now, the file is named jk_constant.rs and it's a bit different. No more dead code and macros to go around the lack of generics. I think this was re-added when you merged master onto your branch.